### DISCUSSION TITLE
How does the dataset active command convert data between -x and no parameters

### DISCUSSION THREAD

**Discussion by wuzeshun997** [openthread/openthread#9081] [Q&A] [answered]
Hi, OpenThread Tcommunity!

sudo ot-ctl dataset active

```
Active Timestamp: 1
Channel: 15
Channel Mask: 0x07fff800
Ext PAN ID: 1111111122222222
Mesh Local Prefix: fd43:d8aa:fe33:c57::/64
Network Key: 00112233445566778899aabbccddeeff
Network Name: OpenThreadDemo
PAN ID: 0x1234
PSKc: 445f2b5ca6f2a93a55ce570a70efeecb
Security Policy: 672 onrc
Done
```
  
sudo ot-ctl dataset active -x
```
0e080000000000010000000300000f35060004001fffe0020811111111222222220708fd43d8aafe330c57051000112233445566778899aabbccddeeff030e4f70656e54687265616444656d6f010212340410445f2b5ca6f2a93a55ce570a70efeecb0c0402a0f7f8
Done
```

I know that the "-x" parameter is encoded in hex-encoded TLV, so I'm trying to parse the same data as above, but there are a few fields that I'm not sure how they were made

This is my own analysis based on some data in [dataset.md](https://github.com/openthread/ot-docs/blob/main/site/en/reference/cli/concepts/dataset.md)
```
0e080000000000010000000300000f35060004001fffe0
Ext PAN ID:         02 - 0x08(8)    - 1111111122222222
Mesh Local Prefix:  07 - 0x08(8)    - fd43d8aafe330c57
Network Key:        05 - 0x10(16)   - 00112233445566778899aabbccddeeff
Network Name:       03 - 0x0e(14)   - 4f70656e54687265616444656d6f
PAN ID:(LE)         01 - 0x02(2)    - 1234
PSKc:               04 - 0x10(16)   - 445f2b5ca6f2a93a55ce570a70efeecb
Security Policy:    0c - 0x04(4)    - 02a0f7f8
```
I don't know which document this Type field is found in  
Just like my parsing data above, I don't know how these parameters are resolved
```
0e080000000000010000000300000f35060004001fffe0
How does this data parse out the Active Timestamp, Channel, and Channel Mask

Active Timestamp: 1
Channel: 15
Channel Mask: 0x07fff800


I can only parse the decimal of 0x02a0 as 672 is the same as here, 
but what does the following 0xf7f8 mean and where does this onrc come from? 
I'm not clear :(
02a0(672) f7f8(?)
Security Policy: 672 onrc
```

So I would like to ask how this data is converted and what is the format of conversion  
I am looking forward to your reply very much. 

Best Regards,
ashun.

**Comment by jwhui** [comment, marked as answer]
[Thread 1.3.0](https://www.threadgroup.org/support#specifications) Section 8.10 defines the TLV encoding.

Implementation is in [src/core/meshcop/meshcop_tlvs.hpp](https://github.com/openthread/openthread/blob/main/src/core/meshcop/meshcop_tlvs.hpp)

> 0e080000000000010000000300000f35060004001fffe0
> How does this data parse out the Active Timestamp, Channel, and Channel Mask

Active Timestamp has Type value 14, which is `0xe`.

**Reply by wuzeshun997** [reply]
Thanks for your help! I will recheck and parse my data according to the document. : )
